lib/puma/launcher.rb

```diff
@@ -184,7 +184,6 @@ def run
       when :exit
         # nothing
       end
-      @binder.close_unix_paths
```

Comment:
:|

Reply:
This reflects a bit of refactoring- this line caused the old parent to delete the Unix socket when shutting down, which in this PR needs to be skipped during a promote operation. The call here is made unnecessary by [this change to Cluster#run](https://github.com/puma/puma/pull/2099/files#diff-c2ca2259d38a531715284b3ec3f4b698R564), which includes a call to [`close_binder_listeners`](https://github.com/puma/puma/pull/2099/files#diff-c2ca2259d38a531715284b3ec3f4b698R582), which also cleans up any Unix sockets - but is skipped during promote.

Some of this refactoring could be extracted to a separate PR without the promote logic.

Reply:
See #2112 for the extracted listener-closing refactoring part of this PR, which can be considered separately from the rest of the promote logic.